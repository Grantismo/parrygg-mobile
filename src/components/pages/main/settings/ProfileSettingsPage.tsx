import React from "react";
import { FieldErrors, RegisterOptions, useForm } from "react-hook-form";
import { View } from "react-native";

import Background from "@/components/base/Background";
import Button from "@/components/base/Button";
import Nav from "@/components/base/navigation/Nav";
import Form from "@/components/forms/Form";
import Input from "@/components/forms/Input";
import tw from "@/lib/tailwind";

type FormData = {
  tag: string;
  firstName: string;
  lastName: string;
  bio: string;
  pronouns: string;
};

const ProfileSettingsPage = () => {
  const validation: RegisterOptions[] = [];
  const errors: FieldErrors<FormData> = {};

  const { handleSubmit, register, setValue } = useForm<FormData>({ errors });

  const onSubmit = (data: FormData) => {};

  return (
    <Background>
      <Nav title="Profile Settings" showBack />
      <View style={tw`grow w-full pb-2`}>
        <Form<FormData> {...{ register, setValue, validation, errors }}>
          <Input name="tag" label="Tag" placeholder="e.g. blorppppp" required />
          <Input
            name="firstName"
            label="First name"
            placeholder="John"
            required
          />
          <Input name="lastName" label="Last name" placeholder="Doe" required />
          <Input name="bio" label="Bio" placeholder="e.g. I love smash" />
          <Input
            name="pronouns"
            label="Pronouns"
            placeholder="e.g. they/them"
          />
          <Button title="Update Profile" onPress={handleSubmit(onSubmit)} />
        </Form>
      </View>
    </Background>
  );
};

export default ProfileSettingsPage;
