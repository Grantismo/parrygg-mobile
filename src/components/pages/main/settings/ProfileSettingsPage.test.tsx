import { describe, expect, it } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";

import ProfileSettingsPage from "@/components/pages/main/settings/ProfileSettingsPage";

describe("ProfileSettingsPage", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ProfileSettingsPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
