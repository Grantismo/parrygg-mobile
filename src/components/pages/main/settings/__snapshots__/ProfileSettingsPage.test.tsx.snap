// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSettingsPage renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1b1b1b",
    }
  }
>
  <View
    style={
      {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "opacity": 0.03,
          "resizeMode": "cover",
        }
      }
    />
  </View>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      [
        {
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
        },
      ]
    }
  >
    <View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(10, 10, 10, 0)",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 12,
            "paddingLeft": 32,
            "paddingRight": 32,
            "paddingTop": 40,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "width": 56,
            }
          }
        />
        <Text
          collapsable={false}
          style={
            {
              "color": "#fff",
              "fontFamily": "Inter_600SemiBold",
              "fontSize": 20,
            }
          }
        >
          Profile Settings
        </Text>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "width": 56,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={5}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#777",
                  },
                  undefined,
                ]
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "height": "100%",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            [
              {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flexGrow": 1,
              "paddingBottom": 8,
              "width": "100%",
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 4,
                  "width": "100%",
                  "zIndex": 0,
                },
                undefined,
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4278979596,
                  4279637526,
                ]
              }
              style={
                {
                  "borderRadius": 14,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Inter_500Medium",
                    },
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "#fff",
                    },
                    {
                      "backgroundColor": "#161616",
                      "borderRadius": 4,
                      "color": "#fff",
                      "fontSize": 12,
                      "fontWeight": "normal",
                      "marginLeft": 16,
                      "paddingBottom": 1.6,
                      "paddingLeft": 7.2,
                      "paddingRight": 7.2,
                      "paddingTop": 1.6,
                      "position": "absolute",
                      "top": -12,
                      "zIndex": 10,
                    },
                  ]
                }
              >
                Tag
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "#fff",
                      },
                      {
                        "color": "#FFC93F",
                        "fontSize": 12,
                        "marginLeft": 4,
                      },
                    ]
                  }
                >
                   *
                </Text>
              </Text>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingBottom": 15,
                      "paddingLeft": 18,
                      "paddingRight": 18,
                      "paddingTop": 13,
                    },
                    false,
                    {
                      "borderRadius": 14,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  blurOnSubmit={false}
                  name="tag"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="e.g. blorppppp"
                  placeholderTextColor="#6F6F6F"
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "color": "#fff",
                        "display": "flex",
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 16,
                      },
                    ]
                  }
                />
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#fff",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 4,
                  "width": "100%",
                  "zIndex": 0,
                },
                undefined,
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4278979596,
                  4279637526,
                ]
              }
              style={
                {
                  "borderRadius": 14,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Inter_500Medium",
                    },
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "#fff",
                    },
                    {
                      "backgroundColor": "#161616",
                      "borderRadius": 4,
                      "color": "#fff",
                      "fontSize": 12,
                      "fontWeight": "normal",
                      "marginLeft": 16,
                      "paddingBottom": 1.6,
                      "paddingLeft": 7.2,
                      "paddingRight": 7.2,
                      "paddingTop": 1.6,
                      "position": "absolute",
                      "top": -12,
                      "zIndex": 10,
                    },
                  ]
                }
              >
                First name
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "#fff",
                      },
                      {
                        "color": "#FFC93F",
                        "fontSize": 12,
                        "marginLeft": 4,
                      },
                    ]
                  }
                >
                   *
                </Text>
              </Text>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingBottom": 15,
                      "paddingLeft": 18,
                      "paddingRight": 18,
                      "paddingTop": 13,
                    },
                    false,
                    {
                      "borderRadius": 14,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  blurOnSubmit={false}
                  name="firstName"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="John"
                  placeholderTextColor="#6F6F6F"
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "color": "#fff",
                        "display": "flex",
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 16,
                      },
                    ]
                  }
                />
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#fff",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 4,
                  "width": "100%",
                  "zIndex": 0,
                },
                undefined,
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4278979596,
                  4279637526,
                ]
              }
              style={
                {
                  "borderRadius": 14,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Inter_500Medium",
                    },
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "#fff",
                    },
                    {
                      "backgroundColor": "#161616",
                      "borderRadius": 4,
                      "color": "#fff",
                      "fontSize": 12,
                      "fontWeight": "normal",
                      "marginLeft": 16,
                      "paddingBottom": 1.6,
                      "paddingLeft": 7.2,
                      "paddingRight": 7.2,
                      "paddingTop": 1.6,
                      "position": "absolute",
                      "top": -12,
                      "zIndex": 10,
                    },
                  ]
                }
              >
                Last name
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "#fff",
                      },
                      {
                        "color": "#FFC93F",
                        "fontSize": 12,
                        "marginLeft": 4,
                      },
                    ]
                  }
                >
                   *
                </Text>
              </Text>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingBottom": 15,
                      "paddingLeft": 18,
                      "paddingRight": 18,
                      "paddingTop": 13,
                    },
                    false,
                    {
                      "borderRadius": 14,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  blurOnSubmit={false}
                  name="lastName"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Doe"
                  placeholderTextColor="#6F6F6F"
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "color": "#fff",
                        "display": "flex",
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 16,
                      },
                    ]
                  }
                />
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#fff",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 4,
                  "width": "100%",
                  "zIndex": 0,
                },
                undefined,
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4278979596,
                  4279637526,
                ]
              }
              style={
                {
                  "borderRadius": 14,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Inter_500Medium",
                    },
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "#fff",
                    },
                    {
                      "backgroundColor": "#161616",
                      "borderRadius": 4,
                      "color": "#fff",
                      "fontSize": 12,
                      "fontWeight": "normal",
                      "marginLeft": 16,
                      "paddingBottom": 1.6,
                      "paddingLeft": 7.2,
                      "paddingRight": 7.2,
                      "paddingTop": 1.6,
                      "position": "absolute",
                      "top": -12,
                      "zIndex": 10,
                    },
                  ]
                }
              >
                Bio
              </Text>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingBottom": 15,
                      "paddingLeft": 18,
                      "paddingRight": 18,
                      "paddingTop": 13,
                    },
                    false,
                    {
                      "borderRadius": 14,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  blurOnSubmit={false}
                  name="bio"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="e.g. I love smash"
                  placeholderTextColor="#6F6F6F"
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "color": "#fff",
                        "display": "flex",
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 16,
                      },
                    ]
                  }
                />
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#fff",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 4,
                  "width": "100%",
                  "zIndex": 0,
                },
                undefined,
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4278979596,
                  4279637526,
                ]
              }
              style={
                {
                  "borderRadius": 14,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Inter_500Medium",
                    },
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "#fff",
                    },
                    {
                      "backgroundColor": "#161616",
                      "borderRadius": 4,
                      "color": "#fff",
                      "fontSize": 12,
                      "fontWeight": "normal",
                      "marginLeft": 16,
                      "paddingBottom": 1.6,
                      "paddingLeft": 7.2,
                      "paddingRight": 7.2,
                      "paddingTop": 1.6,
                      "position": "absolute",
                      "top": -12,
                      "zIndex": 10,
                    },
                  ]
                }
              >
                Pronouns
              </Text>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingBottom": 15,
                      "paddingLeft": 18,
                      "paddingRight": 18,
                      "paddingTop": 13,
                    },
                    false,
                    {
                      "borderRadius": 14,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  blurOnSubmit={false}
                  name="pronouns"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="e.g. they/them"
                  placeholderTextColor="#6F6F6F"
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "color": "#fff",
                        "display": "flex",
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 16,
                      },
                    ]
                  }
                />
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#fff",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 16,
                },
                undefined,
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4293899034,
                  4294953798,
                ]
              }
              style={
                [
                  {
                    "backgroundColor": "rgba(56, 52, 42, 0.38)",
                  },
                  {
                    "borderRadius": 14,
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderWidth": 2,
                      "color": "#000",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 15,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 13,
                    },
                    {
                      "borderColor": "#ffd978",
                    },
                    false,
                    false,
                    false,
                    {
                      "borderRadius": 14,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "#fff",
                      },
                      [
                        {
                          "color": "#000",
                        },
                        false,
                        {
                          "fontSize": 16,
                        },
                      ],
                    ]
                  }
                >
                  Update Profile
                </Text>
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
